pages/docs/reference/whatsnew1420.md

```diff
@@ -0,0 +1,387 @@
+{:.note}
+
+Kotlin 1.4.20 can compile string concatenations into [dynamic invocations](https://docs.oracle.com/javase/7/docs/technotes/guides/vm/multiple-language-support.html#invokedynamic)
+on JVM 9+ targets, therefore improving the performance.
+
+Currently, this feature is experimental and covers the following cases:
+- `String.plus` in the operator (`a + b`), explicit (`a.plus(b)`), and reference (`(a::plus)(b)`) form.
+- `toString` on inline and data classes.
+- string templates except for ones with a single non-constant argument (see [KT-42457](https://youtrack.jetbrains.com/issue/KT-42457)).
+
+To enable `invokedynamic` string concatenation, add the `-Xstring-concat` compiler option with one of the following values:
+- `indy-with-constants` to perform `invokedynamic` concatenation on strings with [StringConcatFactory.makeConcatWithConstants()](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/StringConcatFactory.html#makeConcatWithConstants-java.lang.invoke.MethodHandles.Lookup-java.lang.String-java.lang.invoke.MethodType-java.lang.String-java.lang.Object...-).
+- `indy` to perform `invokedynamic` concatenation on strings with [StringConcatFactory.makeConcat()](https://docs.oracle.com/javase/9/docs/api/java/lang/invoke/StringConcatFactory.html#makeConcat-java.lang.invoke.MethodHandles.Lookup-java.lang.String-java.lang.invoke.MethodType-).
+- `inline` to switch back to the classic concatenation via `StringBuilder.append()`.
+
+[**Back to top**](#)
+
+## Kotlin/JS
+
+Kotlin/JS keeps evolving fast, and in 1.4.20 you can find a number experimental features and improvements:
+
+- [Gradle DSL changes](#gradle-dsl-changes)
+- [New Wizard templates](#new-wizard-templates)
+- [Ignoring compilation errors with IR compiler](#ignoring-compilation-errors-with-ir-compiler)
+
+### Gradle DSL changes
+
+The Kotlin/JS Gradle DSL receives a number of updates which simplify project setup and customization. This includes webpack
+configuration adjustments, modifications to the auto-generated `package.json` file, and improved control over transitive dependencies,
+among others.
+
+#### Single point for webpack configuration
+
+A new configuration block `commonWebpackConfig` is available for the browser target. Inside it, you can adjust common
+settings from a single point, instead of having to duplicate configurations for the `webpackTask`, `runTask`, and `testTask`
+respectively.
+
+To enable CSS support by default for all three tasks, add the following snippet in the `build.gradle(.kts)` of your project:
+
+```groovy
+browser {
+    commonWebpackConfig {
+        cssSupport.enabled = true
+    }
+    binaries.executable()
+}
+```
+
+Learn more about [configuring webpack bundling](js-project-setup.html#configuring-webpack-bundling).
+
+#### package.json customization from Gradle
+
+For more fine-grained control over you Kotlin/JS package management and distribution, now you can add custom
```

Comment:
minor/optional: we could subdivide the sentence into two. It makes it easier to read.